### ISSUE TITLE
Is there a way to start with a tree or subtree expanded?

### ISSUE THREAD

**Issue by BrianGenisio** [wix/angular-tree-control#14]
I like this implementation much better than angular-treeview, but one thing it gave is the ability to start with a tree or subtree expanded.  The data structure included a `collapsed` property that you can set.

I don't see anything like this in this control.  Am I missing something?  Was this excluded on purpose?  If not, any thoughts on the interface for doing this?  (will you accept a PR?)

It seems like your code goes with the concept of 'expanded' instead of 'collapsed'.  I can think of two ways to do this:  
1. A data node can include a 'expanded' property.  If it exists, it sets the default state.  This would be analogous to the way angular-treeview does it.  I don't care for this so much because it is conflating data with UI state.
2. Instead, the `treecontrol` could expose `expandedNodes` via a `$scope` binding.  Doing this would allow the consumer to control the expanded state of the control as well as peek at the expanded state if it so desires.

Thoughts?

**Comment by tdebroc** [comment]
To expand all Nodes at beggining, you can do that:

var allNodes = [];
  function addToAllNodes(children) {
    console.log(children)
    if (!children || typeof(children) == "array" && children.length == 0) {
      return;
    }
    for (var i = 0; i < children.length; i++) {
      allNodes.push(children[i]);
      addToAllNodes(children[i].children);
    }
  }
  addToAllNodes($scope.dataForTheTree);
  
  $scope.expandedNodes = allNodes;

**Comment by dblanchard13** [comment]
`typeof(children) == 'array'` will never return true because using `typeof` on an array will always return `'object'`. In order to check that something is an array you need to use `Array.isArray()`. Though you also don't need to check if it's an array since it will always be an array or `undefined` and the first check you do ensures that it if it's `undefined` the second and third checks won't happen. You can also do this recursively which keeps the flat array scoped within the function:
```
function addToAllNodes(array, flatArray){
  flatArray = flatArray || [];
  if(!children || !children.length){
    return;
  }
  array.forEach(function(node){
    flatArray.push(node);
    addToAllNodes(node.children, flatArray);
  });
  return flatArray;
};

$scope.expandedNodes = addToAllNodes($scope.dataForTheTree);
```

**Comment by MiniNova** [comment]
```
        function addToAllNodes(array, flatArray){
            flatArray = flatArray || [];
            array.forEach(function(node){
                if( !node.children || Object.keys(node.children).length == 0) {
                    return;
                }
                flatArray.push(node);
                addToAllNodes(node.children, flatArray);
            });
            return flatArray;
        };
```

**Comment by mostlybob** [comment]
I would like to use this method to programmatically collapse a node:

``` javascript 
angular.element("#treeview").controller("Hierarchy").collapseNode(source);
```

However, at the controller level, this expression returns undefined. I have the controller defined at the outer container level:

``` html
<div ng-app="myApp" ng-controller="Hierarchy" ng-cloak>
```

I tried adding a controller attribute with this name to the treecontrol element, but no effect. What exactly is the `controller` method on `element` referring to and how do I set up my HTML to provide the reference? Thanks!
